src/fluent.rs

```diff
@@ -160,6 +196,36 @@ impl Localizer {
     }
 }
 
+pub trait MessageKey {
+    fn key(&self) -> &str;
+
+    fn attribute(&self) -> Option<&str> {
+        None
+    }
+}
+
+impl MessageKey for str {
+    fn key(&self) -> &str {
+        self
+    }
+}
```

Comment:
This will give us backwards compatibility yes? Very cool! 👍 Does this also work for `&String`?

Reply:
I will try.. I also though about doing `impl<S: AsRef<str>> MessageKey for S`, that should work as well.

Reply:
(It should be noted that &String only works now because of auto-deref, which should still work.)

Reply:
Nope, does not work, will go with the AsRef version.